### PR TITLE
Issue #2117: Remove site_mail from installer since people misundersta…

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/2117

### CODE REVIEW

**Comment by docwilmot** [comment]
Seems the Devel module folder got committed with this PR.

**Comment by jenlampton** [comment]
hah! :) Pushed a new PR without it (and squashed the commits)

File: core/modules/system/system.install [hunk 1]
```diff
@@ -672,6 +672,48 @@ function system_install() {
...
+ * @return (string)
+ *   The bare domain of the starting URL like example.com or example.co.uk.
+ */
+function backdrop_get_bare_domain($url) {
+  $parts = parse_url($url);
+
+  if ($parts['host']) {
+    $domain = $parts['host'];
+  }
+  else if ($parts['path']) {
```
Comment (quicksketch) [inline core/modules/system/system.install]: Should be an `elseif`

File: core/modules/system/system.install [hunk 2]
```diff
@@ -672,6 +672,48 @@ function system_install() {
...
+/**
+ * Take a full URL and return only the bare domain.
+ *
+ * @param (string) $url
+ *   A fully-qualified URL like https://www.example.co.uk/path-to-page.
+ *
+ * @return (string)
+ *   The bare domain of the starting URL like example.com or example.co.uk.
+ */
+function backdrop_get_bare_domain($url) {
```
Comment (quicksketch) [inline core/modules/system/system.install]: If this is a general utility function, it should probably live in `common.inc` with other `backdrop_*` functions.

Comment (quicksketch) [reply core/modules/system/system.install]: We also need new testing to cover this function with various inputs such as with or without www, two-letter domain names and TLDs, multiple subdomains, etc.

File: core/scripts/install.sh [hunk 3]
```diff
@@ -138,7 +138,6 @@ $settings = array(
     ),
     'install_configure_form' => array(
       'site_name' => $options['site-name'],
-      'site_mail' => $options['site-mail'],
```
Comment (quicksketch) [inline core/scripts/install.sh:141]: Looks like this will break the `--site-mail` option that is passed into `install.sh`. It should either still be accommodated, or removed from the list of options.

File: core/includes/install.core.inc [hunk 4]
```diff
@@ -547,25 +546,25 @@ function install_tasks($install_state) {
   // to the installation profile.
   $tasks = array(
     'install_select_language' => array(
-      'display_name' => st('Choose language'),
+      'display_name' => st('Choose your language'),
```
Comment (quicksketch) [inline core/includes/install.core.inc]: I'm not sure about including the word "your" here. That makes it sound like it's the language of the user doing the installing, not necessarily the language the site itself will use. Can we revert this change and save it for a separate issue?

File: core/includes/install.core.inc [hunk 5]
```diff
@@ -547,25 +546,25 @@ function install_tasks($install_state) {
...
       'run' => INSTALL_TASK_RUN_IF_REACHED,
     ),
     'install_select_profile' => array(
-      'display_name' => st('Choose profile'),
+      'display_name' => st('Choose your profile'),
```
Comment (quicksketch) [inline core/includes/install.core.inc]: Same with all other additions of "your", probably better a separate issue.

**Review by quicksketch — CHANGES_REQUESTED** [review]

File: core/includes/install.core.inc (hunk 5 above)
Comment (jenlampton) [reply core/includes/install.core.inc]: done.

File: core/includes/install.core.inc (hunk 4 above)
Comment (jenlampton) [reply core/includes/install.core.inc]: reverted.

File: core/modules/system/system.install (hunk 1 above)
Comment (jenlampton) [reply core/modules/system/system.install]: fixed.

File: core/modules/system/system.install (hunk 2 above)
Comment (jenlampton) [reply core/modules/system/system.install]: moved. still needs tests.

File: core/scripts/install.sh (hunk 3 above)
Comment (jenlampton) [reply core/scripts/install.sh:141]: removed.

**Comment by jenlampton** [comment]
Added new tests for backdrop_get_bare_domain() and changed the testing profile while I was at it.

**Comment by quicksketch** [comment]
I added more tests to backdrop_get_bare_domain, made it act consistently in regards to subdomains, and fixed the documentation. New PR at https://github.com/backdrop/backdrop/pull/1702.
